### PR TITLE
fix: Prevent NOPs on allacation and resetting

### PR DESCRIPTION
**Pull request by bingen**

### CODE REVIEW

File: src/Governance.sol [hunk 1]
```diff
@@ -641,12 +642,14 @@ contract Governance is Multicall, UserProxyFactory, ReentrancyGuard, Ownable, IG
...
         // To ensure the change is safe, enforce uniqueness
-        _requireNoDuplicates(_initiatives);
         _requireNoDuplicates(_initiativesToReset);
+        _requireNoDuplicates(_initiatives);
 
         // Explicit >= 0 checks for all values since we reset values below
         _requireNoNegatives(_absoluteLQTYVotes);
         _requireNoNegatives(_absoluteLQTYVetos);
+        // If the goal is to remove all votes from an initiative, including in _initiativesToReset is enough
+        _requireNoNOP(_absoluteLQTYVotes, _absoluteLQTYVetos);
```
Comment (danielattilasimon) [inline src/Governance.sol:652]: Technically, this renders the previous `_requireNoNegatives()` checks redundant, although we might want to revise the name and revert message of `_requireNoNOP` if we eliminate them.

File: src/utils/UniqueArray.sol [hunk 2]
```diff
@@ -5,6 +5,8 @@ pragma solidity ^0.8.24;
 /// @param arr - List to check for dups
 function _requireNoDuplicates(address[] calldata arr) pure {
     uint256 arrLength = arr.length;
+    if (arrLength == 0) return;
```
Comment (danielattilasimon) [inline src/utils/UniqueArray.sol:8]: Nice catch!

File: src/Governance.sol (hunk 1 above)
Comment (bingen) [reply src/Governance.sol:652]: Why? `_requireNoNOP` doesn’t prevent a negative value, only that for each pair at least one is positive.
Anyway, I was thinking we would remove `_requireNoNegatives` when we change the signature `int88` to unsigned. It’s weird to call them absolute but allow them to be negative to then check they are not.

Comment (danielattilasimon) [reply src/Governance.sol:652]: Ah yes, I forgot it's `||`.

Agree the interface should be unsigned.

**Pull request merged**